src/components/CommonComponents/Pagination/index.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {
+  createElement,
+  FC,
+  MouseEvent,
+  ReactElement,
+  useCallback,
+  useMemo,
+} from 'react';
+import classnames from 'classnames';
+import { buildPaginationModel } from './utils/buildPaginationModel';
+import { buildComponentData } from './utils/buildComponentData';
+import styles from './index.module.scss';
+
+interface UsePaginationPagesParameters {
+  pageCount: number;
+  currentPage: number;
+  onPageChange?: (e: MouseEvent, n: number) => void;
+  hrefBuilder: (n: number) => string;
+  marginPageCount: number;
+  showPages?: boolean;
+  surroundingPageCount: number;
+}
+
+interface Props {
+  currentPage: number;
+  hrefBuilder?: (n: number) => string;
+  marginPageCount: number;
+  pageCount: number;
+  onPageChange?: (e: MouseEvent, n: number) => void;
+  showPages?: boolean;
+  surroundingPageCount?: number;
+  wrapperClassName?: string;
+}
+
+function usePaginationPages({
```

Comment:
Also, Consider using `const` instead of `function` in React files.

Reply:
fixed